### ISSUE TITLE
Nested types (flatten args)

### ISSUE THREAD

**Issue by Alxandr** [murarth/gumdrop#21]
It would be really nice if the following worked, in order to enable re-use of options between multiple commands:

```rust
#[derive(Options)]
struct CommonOpts {
  foo: bool,
}

#[derive(Options)]
struct Opts {
  bar: bool,

  #[options(flatten)]
  common: CommonOpts,
}

// call using `program --foo --bar`
```

**Comment by murarth** [comment]
This would be nice, but it's looking rather tricky to implement given the code currently generated for `Options::parse`, which is kind of a mess ([see here](https://github.com/murarth/gumdrop/blob/decfb7567540c5365b3cfa01148a470cae90cb32/gumdrop_derive/src/lib.rs#L582-L615)).

However, I'd like to keep this issue open to remind me to implement this feature as part of a future refactor.

**Comment by Alxandr** [comment]
Yeah. I tried doing manually (ie, not using derive), but I gave up mostly due to `&'static str` return types all over the place. Went back to structopt for now.
